src/clj/puppetlabs/services/ca/certificate_authority_core.clj

```diff
@@ -56,11 +58,27 @@
       ;; Respond to all CSR validation failures with a 400
       (middleware-utils/plain-response 400 msg))))
 
+      ;;A recipient MUST ignore If-Modified-Since if the request contains an
+      ;; If-None-Match header field;
 (defn handle-get-certificate-revocation-list
-  [{:keys [cacrl]}]
-  (-> (ca/get-certificate-revocation-list cacrl)
-      (rr/response)
-      (rr/content-type "text/plain")))
+  "Always return the crl if no 'if-modified-since' header is provided or if that header is not in correct http-date format.
+  If the header is present and correctly formatted, only return the crl if the master crl is newer than the agent crl."
+  [request {:keys [cacrl]}]
+  (let [agent-crl-last-modified-val (rr/get-header request "If-Modified-Since")
+        agent-crl-last-modified-date-time (try
+                                            (time-format/parse
+                                              (time-format/formatters :rfc822)
+                                              (string/replace agent-crl-last-modified-val #"GMT" "+0000"))
+                                            (catch Exception e
+                                              "unknown"))
+        master-crl-last-modified-date-time (ca/get-crl-last-modified cacrl)]
+    (if (or (= "unknown" agent-crl-last-modified-date-time) (time/after? master-crl-last-modified-date-time agent-crl-last-modified-date-time))
```

Comment:
is "unknown" too hacky here?  I could just return nil and check for the presence of the variable (nil?), but I kind of like "unknown" since its clearer what's going on.